ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadQueue.swift

```diff
@@ -424,30 +424,20 @@ class ResourceDownloadQueue: HTTPDownloadDelegate {
 
         // TEMP:  wrap the newly-downloaded resources with a kmp.json.
         //        Serves as a bridge until we're downloading actual .kmps for keyboards.
-        let wrappedKeyboards = Migrations.migrateToKMPFormat(keyboards)
-
-        wrappedKeyboards.forEach { keyboard in
-          ResourceFileManager.shared.addResource(keyboard)
-        }
+        let _ = Migrations.migrateToKMPFormat(keyboards)
 
         if let package: KeyboardKeymanPackage = ResourceFileManager.shared.getInstalledPackage(for: keyboards[0]) {
           batch.completionBlock?(package, nil)
         } else {
           log.error("Could not load metadata for newly-installed keyboard")
         }
       }
+      // else "if resource is installed from an actual KMP"
+      // - in other words, the long-term target.
     } else if let batch = batch as? DownloadBatch<InstallableLexicalModel> {
       let task = batch.downloadTasks[0] // It's always at this index.
-      let (lm, package) = installLexicalModelPackage(downloadedPackageFile: URL(fileURLWithPath: task.request.destinationFile!))
-      if let lm = lm, let package = package {
-        if !isUpdate {
-          // Since we don't generate the notification as above, we need to manually update
-          // the lexical model's metadata.
-          Manager.shared.updateUserLexicalModels(with: lm)
-        }
-
-        batch.completionBlock?(package, nil)
-      }
+      let packagePath = URL(fileURLWithPath: task.request.destinationFile!)
+      batch.completeWithPackage(fromKMP: packagePath)
```

Comment:
So, so smooth how this resolves within ResourceDownloadQueue now.  Once we download KMPs for keyboards, this function's `if`-`else` ladder can be dropped entirely, greatly simplifying this part of the code.